### ISSUE TITLE
Quality degradation on large images with Klein-9B Base

### ISSUE THREAD

**Issue by iqddd** [Haoming02/sd-webui-forge-classic#786]
When generating at resolutions larger than 1024x1024 (tested 1216x1216, 1408x1024, 1024x1408), I'm seeing a ton of artifacts showing up: texture defects, doll-like anatomy, overall loss of sharpness, broken colors (overexposure, too much contrast), objects merging together (locally), messed up pose physics, background distortions/blur/noise.

Tested with CFG=4, Sampler Euler, Steps=50
Tried different Schedulers, other CFG values and samplers. Bottom line – it's the resolution that's causing this.

ps: Klein-9B suffers from this too.
ps2: no issues in ComfyUI or in the bfl.ai sandbox
ps3: No tiled VAE

<details><summary>Example</summary>
1024x1024
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/d794eb71-f48d-43b4-922a-cc3a962da30f" />
1408x1024
<img width="1408" height="1024" alt="Image" src="https://github.com/user-attachments/assets/21b6f520-7809-4871-9d88-27f5fb19f6c5" />
1216x1216
<img width="1216" height="1216" alt="Image" src="https://github.com/user-attachments/assets/a83a2e6b-a8b6-4306-8434-eccc85ee554f" />
</details>

**Comment by Haoming02** [comment]
> no issues in ComfyUI

Did you try the exact same Seed?

**Comment by iqddd** [comment]
No. Different seeds.
<details><summary>Same seed=0</summary>
1024x1024
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/beffe086-2e1a-4bea-86c0-5864cbc420f7" />
1408x1024
<img width="1408" height="1024" alt="Image" src="https://github.com/user-attachments/assets/e1b01611-af6a-4679-be10-3b2683b4f66c" />
1216x1216
<img width="1216" height="1216" alt="Image" src="https://github.com/user-attachments/assets/96e568c1-6e64-4946-8194-3bdd9e85f198" />
</details>

ps: Pretty sure it's somehow related to scale/shift rope. At least that's what GPT says. xD
I don't really get it myself tbh.

**Comment by Haoming02** [comment]
I meant comparing the images between Forge and ComfyUI with the exact same parameters and Seed...

> Pretty sure it's somehow related to scale/shift rope

Cool. Forge's implementation is exactly the same as ComfyUI's since I simply ~~stole~~ copied them

**Comment by iqddd** [comment]
Ok. I'll do Forge vs. Comfy in a bit
ps: I'm just not sure if forge seed=0 is the same thing as comfy seed=0

**Comment by iqddd** [comment]
<details><summary>comfy seed=0</summary>
1024x1024
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/7ce0db3c-d2c4-4400-abc7-696f93ff9871" />
1408x1024
<img width="1408" height="1024" alt="Image" src="https://github.com/user-attachments/assets/6ba46cd2-ab97-4314-b196-0f6ec2eeff1b" />
1216x1216
<img width="1216" height="1216" alt="Image" src="https://github.com/user-attachments/assets/5c3a4b3d-947b-4c47-abe4-45dcbed5ad68" />
</details>

**Comment by LoneVit** [comment]
This thread is kinda summarize my feelings from working with Klein-9B Base in Forge and why I stopped trying to use it in Forge for a while. Sometimes i got good results, but mostly they were unexpected and weird, insane oversaturation was most noticeable issue for me. So I stick to Z-image Base, which works much better in Forge. But looking on comparing pictures from Comfy, they really sell Comfy... I thinks maybe I should even try ZIB in Comfy too, cos ZIB results sometimes were far from perfect, but I guessed it is just how model works (and probably it is), though don't really like to work in Comfy, but image quality is priority for me.

**Comment by zeydsama** [comment]
I’ve been encountering a similar behavior. I noticed some over-saturation with Klein 9B in txt2img; it’s subtle enough to be tolerable upon closer inspection, but definitely present. However, the effect becomes much more pronounced and significant when using img edit.
I’m curious, could this be a characteristic side effect of the quantized (Q3 K M) models, or is it perhaps related to how the pipeline handles latents in img edit(?)🤔

**Comment by iqddd** [comment]
I only notice the problem when exceeding 1024x1024.
1024x1024 looks great, 1024x1088 already has artifacts.
I'm trying to figure it out, comparing it with the implementation in Comfy, but so far without success.

**Comment by LoneVit** [comment]
> I only notice the problem when exceeding 1024x1024. 1024x1024 looks great, 1024x1088 already has artifacts. 

You posted really good example images, they show very good that it is already not great even in 1024x1024. Just look at color rendering, it is already shifted, and this emphasized even more in other resolutions, so become obvious even to untrained eye. 
Myself I initially tried to generate in 896x1152 and was terrified how bad and messed all details was rendered. Then I tried in 1344x1728 and it was better in details, but very often produced this strange color shifts and over-saturations, so images just looked terrible for a new model.

**Comment by Haoming02** [comment]
*Should* :tm:, *Finally* :tm:, and *Hopefully* :tm:, be fixed~

**Comment by Haoming02** [comment]
~~Umm... did you actually test it?~~

**Comment by iqddd** [comment]
Yep.

**Comment by Haoming02** [comment]
Oh OK XD

**Comment by iqddd** [comment]
I only tested `attention_mask[-1] = 0`. It fixes the issue. Sigma matching seems secondary, at least from what I can tell.

**Comment by LoneVit** [comment]
Made a quick test with previously generated image, need a further testing, but at first sight seems fine now.

**896x1152**
![Image](https://github.com/user-attachments/assets/e991591a-d8db-4d21-8a55-04a08e98d193)

@Haoming02 Thank you for all your efforts with this project, much appreciated 🙏

**Comment by zeydsama** [comment]
It works!! No more oversaturation in image edits either! Thank you so much! I hope you really enjoy this project, and as always, no rush and please take your time! @Haoming02

**Comment by zeydsama** [comment]
sorry for the ping @Haoming02 . i have encountered the over-saturated issue again on img edits using klein. any changed configurations especially klein's?🙏🥲

**Comment by Haoming02** [comment]
I... don't think so
